doc/htmldoc/conf.py

```diff
@@ -293,34 +293,25 @@ def add_button_to_examples(app, env, docnames):
         if file.stem == "index" or file.stem == "hpc_benchmark":
             continue
 
-        get_path = file.parent
-        get_subdir = get_path.stem
-
-        with open(file, "r") as f:
-
-            lines = f.readlines()
-
+        with open(file, "r+") as f:
+            data = f.readlines()
```

Comment:
Maybe read the lines at the bottom

Reply:
sorry what do you mean?

Reply:
I just moved it a few lines down, which I think is what you meant?

Reply:
I mean move opening of the file and the reading to the bottom after computing the `prolog` variable. Because you don't need to open the file before that.

Reply:
It looks better now, thank you.